src/args.rs

```diff
@@ -174,7 +181,11 @@ impl Args {
     }
 
     pub(crate) fn get_project_file_walker_types(&self) -> Types {
-        get_project_file_walker_types(self.language)
+        get_project_file_walker_types(self.language.map(|language| vec![language]).or_else(|| {
+            self.query_per_language
```

Comment:
This is basically asserting that if you pass `query_per_language` those are "all" of the languages that you're interested in

Some of this is probably a little bit conceptually unresolved but that seems more or less like the right idea